### PR TITLE
Update default hyperparameters

### PR DESCRIPTION
**Pull request by JackyP**
The purpose of this change is to improve performance of the model on default hyperparameters, particularly for TabNet.

Update TabNet with:
* ``num_groups`` default to 1 - see https://github.com/mlr-org/mlr3keras/issues/19.
* ``optimizer`` to an exponential decay learn rate scheduler and Adam optimizer with learn_rate = 0.02, decay = 0.9 based on typical hyperparameters from the tabnet original paper https://arxiv.org/pdf/1908.07442.pdf, page 15.

Updates LearnerRegKeras and LearnerClassifKeras:
 * Increases default epochs to 100

Also adds tensorflow as a dependency due to the exponential decay learn rate scheduler.

### CODE REVIEW

**Comment by pfistfl** [comment]
Hey, thanks a lot. 

Can we maybe document the `num_groups` thing somewhere?

**Comment by pfistfl** [comment]
Merging, seems alright.
